### PR TITLE
[VFS] Maximum disk space and usage

### PR DESCRIPTION
**Pull request by jinroh**

### CODE REVIEW

File: cmd/instances.go [hunk 1]
```diff
@@ -235,6 +247,7 @@ func init() {
 	addInstanceCmd.Flags().StringVar(&flagTimezone, "tz", "", "The timezone for the user")
 	addInstanceCmd.Flags().StringVar(&flagEmail, "email", "", "The email of the owner")
 	addInstanceCmd.Flags().StringVar(&flagPublicName, "public-name", "", "The public name of the owner")
+	addInstanceCmd.Flags().StringVar(&flagDiskSpace, "disk-space", "", "The size of the allowed")
```
Comment (nono) [inline cmd/instances.go]: Maybe we can call this disk-quota instead of disk-space

File: cmd/instances.go [hunk 2]
```diff
@@ -235,6 +247,7 @@ func init() {
 	addInstanceCmd.Flags().StringVar(&flagTimezone, "tz", "", "The timezone for the user")
 	addInstanceCmd.Flags().StringVar(&flagEmail, "email", "", "The email of the owner")
 	addInstanceCmd.Flags().StringVar(&flagPublicName, "public-name", "", "The public name of the owner")
+	addInstanceCmd.Flags().StringVar(&flagDiskQuota, "disk-space", "", "The size of the allowed")
```
Comment (aenario) [inline cmd/instances.go]: Lack a word there, no ?

File: pkg/vfs/errors.go [hunk 3]
```diff
@@ -35,4 +35,6 @@ var (
 	ErrDirNotEmpty = errors.New("Directory is not empty")
 	// ErrWrongCouchdbState is given when couchdb gives us an unexpected value
 	ErrWrongCouchdbState = errors.New("Wrong couchdb reduce value")
+	// ErrFileTooBig is used when there is no more space left on the filesystem
+	ErrFileTooBig = errors.New("The file is too big")
```
Comment (aenario) [inline pkg/vfs/errors.go]: May be we can make this error a bit more clear "Adding file would exceed quota", as it is now it makes me think more of 413 error, aka "we will never accept a file bigger than 10Go"

File: docs/settings.md [hunk 4]
```diff
@@ -74,6 +76,8 @@ Content-type: application/vnd.api+json
     "type": "io.cozy.settings",
     "id": "io.cozy.settings.disk-usage",
     "attributes": {
+      "is_limited": true,
+      "total_size": "123456789",
```
Comment (nono) [inline docs/settings.md]: I'd have called it `quota` too

Comment (nono) [inline docs/settings.md]: And I'd have not put a `is_limited` field. If true, there is a `quota`/`total_size`. If false, we just omit this field.

File: pkg/instance/instance.go [hunk 5]
```diff
@@ -72,6 +72,8 @@ type Instance struct {
 	Locale string `json:"locale"`         // The locale used on the server
 	Dev    bool   `json:"dev"`            // Whether or not the instance is for development
 
+	BytesDiskQuota int64 `json:"disk_quota,string"` // The total size in bytes allowed to the user
```
Comment (nono) [inline pkg/instance/instance.go]: `omitempty`?

File: pkg/instance/instance.go [hunk 6]
```diff
@@ -191,6 +195,11 @@ func (i *Instance) AppsFS(appsType apps.AppType) afero.Fs {
 	panic(fmt.Errorf("Unknown application type %s", string(appsType)))
 }
 
+// DiskQuota returns the number of bytes allowed on the disk to the user.
+func (i *Instance) DiskQuota() (int64, error) {
+	return i.BytesDiskQuota, nil
```
Comment (nono) [inline pkg/instance/instance.go]: Why do you return an error if it is always nil?

Comment (jinroh) [inline pkg/instance/instance.go]: It implements an interface `DiskThresholder` which defines a `DiskQuota() (int64, error)`, allowing the possibility to have an error... maybe that is pretty useless for now...

File: pkg/instance/instance.go (hunk 5 above)
Comment (jinroh) [inline pkg/instance/instance.go]: why not. done.

File: pkg/instance/instance.go (hunk 6 above)
Comment (nono) [inline pkg/instance/instance.go]: Yes, but the implementation in the test also return nil for the error

Comment (jinroh) [inline pkg/instance/instance.go]: I feel it is premature now that you noted that. I removed the error return from the function's signature.

**Comment by jinroh** [comment]
rebased on master

**Pull request merged**